drivers/w1/w1_ds2477_85_common.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2022 Thomas Stranger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common functions for Maxim DS2477,DS2485 1-Wire Masters
+ */
+
+#include "w1_ds2477_85_common.h"
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/w1.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(w1_ds2477_85, CONFIG_W1_LOG_LEVEL);
+
+int ds2477_85_write_port_config(const struct device *dev, uint8_t reg, uint16_t value)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	uint8_t buf[5] = {CMD_WRITE_W1_PORT_CONFIG, 3, reg};
+	int ret;
+
+	__ASSERT_NO_MSG(reg <= PORT_REG_COUNT);
+
+	sys_put_le16(value, &buf[3]);
+	ret = i2c_write_dt(&cfg->i2c_spec, buf, 5);
+	if (ret != 0) {
+		return -EIO;
+	}
+
+	k_usleep(cfg->t_op_us);
+
+	ret = i2c_read_dt(&cfg->i2c_spec, buf, 2);
+	if (ret != 0 || buf[0] != 1 || buf[1] != RESULT_SUCCESS) {
```

Comment:
To be more MISRA-ish

```suggestion
	if ((ret != 0) || (buf[0] != 1) || (buf[1] != RESULT_SUCCESS)) {
```

Reply:
Also, should it be `ret < 0` instead?

Reply:
Changed these comparisons accordingly.